backend/graphql/handler.ts

```diff
@@ -1,25 +1,69 @@
 import { executeQuery } from "../psql/queryExecute";
-import { QUERY_GET_ALL_DAOS } from "../psql/query";
-import { DaoType } from "./types";
-import { DaoItemType } from "../types";
-const { GraphQLObjectType, GraphQLList } = require("graphql");
+import { QUERY_GET_DAOS_COUNT, getPaginatedQuery } from "../psql/query";
+import { DaosAndPageInfoType } from "./types";
+import { DaoItemType, DaoArgType, DaoAndPageResType } from "../types";
+const { GraphQLObjectType, GraphQLInt, GraphQLNonNull } = require("graphql");
+
+const validateRes = (totalDaosRes: any, paginateRes: any, args: DaoArgType) => {
+	return (
+		totalDaosRes &&
+		totalDaosRes.rows &&
+		paginateRes &&
+		paginateRes.rows &&
+		paginateRes.rows.length &&
+		args
+	);
+};
+
+const getDaoPageRes = (
+	totalDaos: number,
+	daos: [DaoItemType],
+	args: DaoArgType
+): DaoAndPageResType => {
+	const pageNumber = args.pageNumber; // page number = 1..n
+	const pageSize = args.pageSize;
+	const hasNext = !(pageNumber * pageSize >= totalDaos); // check if user's total page iterated is >= totalDaos
+	const hasPrev = !(pageNumber <= 1); // check if user is on first page
+	return {
+		daos,
+		pageInfo: {
+			hasNext,
+			hasPrev,
+			totalDaos,
+		},
+	};
+};
 
 export const queryRoot = new GraphQLObjectType({
 	name: "Query",
 	fields: () => ({
-		Daos: {
-			type: new GraphQLList(DaoType),
-			description: "Dao Records",
-			resolve: async () => {
-				const res = await executeQuery(QUERY_GET_ALL_DAOS);
-				if (res && res.rows && res.rows.length) {
-					res.rows.map((item: DaoItemType) => {
+		DaoAndPage: {
+			type: DaosAndPageInfoType,
+			description: "Returns Dao Records and Page Info",
+			args: {
+				pageNumber: {
+					type: new GraphQLNonNull(GraphQLInt),
+					description: "Page number as an argument",
+				},
+				pageSize: {
+					type: new GraphQLNonNull(GraphQLInt),
+					description: "Offset as an argument.",
+				},
+			},
+			resolve: async (_: object, args: DaoArgType) => {
+				const totalDaosRes = await executeQuery(QUERY_GET_DAOS_COUNT);
+				const paginateRes = await executeQuery(
+					getPaginatedQuery(args.pageNumber, args.pageSize)
+				);
+				if (validateRes(totalDaosRes, paginateRes, args)) {
+					const totalDaos = totalDaosRes.rows[0].count || 0;
+					paginateRes.rows.map((item: DaoItemType) => {
 						if (item.app_params) {
 							item.app_params = JSON.stringify(item.app_params);
```

Comment:
why do we call `stringify` here?

Reply:
app_param field in DB has JSONB as type. GraphQL was not having this type so I converted this into string format and returned it. This way it will match the return type.
Check this: https://github.com/scale-it/sigmadao-ui/blob/4a81c8b8f889157283d4d481055e9387f25cdbce/backend/graphql/types.ts#L15
It returns return type in string format.

Reply:
then you should convert it to an object, we don't want to send stringified objects.

Reply:
This will be tackled in postgraphile task that you have created.